src/redux/reducers/currentProjectReducer.ts

```diff
@@ -21,34 +21,32 @@ export const reducer = (state: IProject = null, action: AnyAction): IProject =>
             return null;
         case ActionTypes.LOAD_PROJECT_SUCCESS:
             return { ...action.payload };
-        case ActionTypes.LOAD_PROJECT_ASSETS_SUCCESS:
-            if (state) {
-                const currentAssets = { ...state.assets } || {};
-                action.payload.forEach((asset) => {
-                    if (!currentAssets[asset.id]) {
-                        currentAssets[asset.id] = asset;
-                    }
-                });
-
-                return {
-                    ...state,
-                    assets: currentAssets,
-                };
-            } else {
+        case ActionTypes.SAVE_ASSET_METADATA_SUCCESS:
+            if (!state) {
                 return state;
             }
-        case ActionTypes.SAVE_ASSET_METADATA_SUCCESS:
-            if (state) {
-                const updatedAssets = { ...state.assets } || {};
-                updatedAssets[action.payload.asset.id] = { ...action.payload.asset };
 
-                return {
-                    ...state,
-                    assets: updatedAssets,
-                };
-            } else {
+            const updatedAssets = { ...state.assets } || {};
+            updatedAssets[action.payload.asset.id] = { ...action.payload.asset };
+
+            return {
+                ...state,
+                assets: updatedAssets,
+            };
+        case ActionTypes.SAVE_CONNECTION_SUCCESS:
```

Comment:
If the connection being updated is in use by the current project then also update the current project source/target connection